README.md

```diff
@@ -1 +1,81 @@
-# Android Error Handlesr
\ No newline at end of file
+## Structure
+  The library consist of 2 modules:
+  - **error-handler-core** - defines the DataLayerException, the root exception for an entire data layer, and ResultWrapper, wrapper class based on Railway Oriented Programming pattern.
+  - **error-handler-rest** - which provides error handling for REST API consumption. Note that this module is built on top of [retrofit-error-wrapping](https://github.com/halcyonmobile/retrofit-error-wrapping) library. This module provides subclasses for DataLayerException with possible failures that can occur during an HTTP call.
+
+## Setup
+  ###### Latest stable version is `0.1.0`
+  - Add `jcenter` to your project `build.gradle`
+   ```kotlin 
+repositories {
+       // other repositories
+       jcenter()
+}
+   ``` 
+   - Add this library to your project
+   ```kotlin
+dependencies {
+       // other dependencies
+       implementation "com.halcyonmobile.error-handler:core:<latest_version>"
+       implementation "com.halcyonmobile.error-handler:rest:<latest_version>"
+   }
+   ```
+   - Use the `RestCallAdapter.Builder` and build your own `CallAdapterFactory` for Retrofit2
+   ```kotlin
+val callAdapterFactory = RestCallAdapterFactory.Builder()
+       /* Customize it here. Check his documentation for more */
+       .build()
+   ```
+   - Add it to `Retrofit`
+   ```kotlin
+Retrofit.Builder()
+        .addCallAdapterFactory(callAdapterFactory)
+        /* other setups */
+        .build()
+   ```
+   - Build an `ApiErrorModel`, that represents the model of error that's being returned by your backend, in case something goes wrong.
+   
+   - Mark your retrofit methods with @ParsedError, passing your custom ApiErrorModel from the previous step (*more about this on **How it works** section below*)
+   ```kotlin
+   interface FoodService {
+       
+       //Tip! @ParsedError means this method will throw RemoteExceptions,
+       //so it's highly recommended to also add @Throw(RemoteException::class)
+       @ParsedError(MyApiErrorModel::class)
+       @POST("food/order")
+       suspend fun orderFood()
+   }
+   ```
+   
+   - That's it! No more hand-done conversions!
+
+## How it works
+   - When you annotate your retrofit call with the @ParsedError annotation, any exception that will be thrown be it will be automatically converted to a RemoteException type. If you receive an error body from your API, the library will try to parse the body into your model class provided in the ParsedError and if the parsing succeeds, then it will throw an ApiError with the parsed body (you will have to cast to your model class when you want to retrieve it).
```

Comment:
"_any exception that will be thrown, will be automatically_", without the _be it_